### PR TITLE
Add validation to contact details

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

Currently, candidates can enter whatever they want for all the fields for contact details.

### Changes proposed in this pull request

This PR adds:
-  a `PostcodeValidator` which is borrowed from [Find](https://github.com/DFE-Digital/manage-courses-backend/blob/5a5838f8644c10fb9d71fa1d5e7f338b648d479a/app/validators/postcode_validator.rb)
- a `PhoneNumberValidator` which is borrowed from [Find](https://github.com/DFE-Digital/manage-courses-backend/blob/5a5838f8644c10fb9d71fa1d5e7f338b648d479a/app/validators/phone_validator.rb) 
- validation for all address fields
- validation for phone number

### Screenshots

image

image

image

### Guidance to review

Get the app running and try putting invalid data.

- [x] Be able to add phone number and review it (#363) 
- [x] Be able to add address and review it (#375)
- [ ] Add validation for phone number (this PR)
- [ ] Add validation for address (this PR)
- [ ] Be able to change your contact details
- [ ] Add completed label to contact details section and show contact details at application review

### Link to Trello card

[191 - Providing contact details](https://trello.com/c/h2MIJjei/191-providing-contact-details)

### CODE REVIEW

**Comment by fofr** [comment]
It doesn't look like it was added in this PR, but error summaries need to be above the title (but after a back link or breadcrumb): http://design-system.service.gov.uk/components/error-summary/#where-to-put-the-error-summary

File: app/models/candidate_interface/contact_details_form.rb [hunk 1]
```diff
@@ -5,6 +5,15 @@ class ContactDetailsForm
     attr_accessor :phone_number, :address_line1, :address_line2, :address_line3,
                   :address_line4, :postcode
 
+    validates :address_line1, :address_line3, :postcode, presence: true, on: :address
+
+    validates :address_line1, :address_line2, :address_line3, :address_line4,
+              length: { maximum: 200 }, on: :address
```
Comment (tvararu) [inline app/models/candidate_interface/contact_details_form.rb]: Can you double check the API docs? I believe they may have a smol-er limit, like 100 iirc

Comment (RobertM905) [reply app/models/candidate_interface/contact_details_form.rb]: Currently the each address field has maxLength of 50. 
I'm fine with increasing this if needs be 👍

**Comment by chubberlisk** [comment]
> It doesn't look like it was added in this PR, but error summaries need to be above the title (but after a back link or breadcrumb): http://design-system.service.gov.uk/components/error-summary/#where-to-put-the-error-summary

Ah, I've updated this for contact details and other forms. Screenshots in description updated too.

File: app/models/candidate_interface/contact_details_form.rb (hunk 1 above)
Comment (chubberlisk) [reply app/models/candidate_interface/contact_details_form.rb]: I'll update it to match the API spec.

**Pull request merged**